components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/user/OpenIDConnectUserEndpointTest.java

```diff
@@ -0,0 +1,160 @@
+
+        };
+    }
+
+    /**
+     *
+     * @param authResponse
+     * @param errorMessage
+     * @param errorCode
+     * @param expectedStatus
+     * @throws Exception
+     */
+    @Test(dataProvider = "provideDataForGetUserClaims")
+    public void testGetUserClaims(String authResponse, String errorMessage, String errorCode, int expectedStatus) throws Exception {
+
+        String clientID = "rgfKVdnMQnJlSSr_pKFTxj3apiwYa";
+
+        UserInfoEndpointException ex = new UserInfoEndpointException(errorCode, errorMessage);
+        Class<?> clazz = OpenIDConnectUserEndpoint.class;
+        Object setHandleError = clazz.newInstance();
+        Method handleError = setHandleError.getClass().
+                getDeclaredMethod("handleError", UserInfoEndpointException.class);
+        handleError.setAccessible(true);
+        Response errorResponse = (Response)
+                handleError.invoke(setHandleError, ex);
+
+        assertEquals(errorResponse.getStatus(), expectedStatus,"Error response values are not same");
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oauthServerConfigurationMock);
+        when(oauthServerConfigurationMock.getTimeStampSkewInSeconds()).thenReturn(3600L);
+        when(userInfoResponseBuilder.getResponseString(tokenResponse)).thenReturn(authResponse);
+        when(userInfoEndpointConfig.getUserInfoResponseBuilder()).thenReturn(userInfoResponseBuilder);
+
+        mockStatic(OAuth2Util.class);
+        when(OAuth2Util.getTenantDomainOfOauthApp(appDO)).thenReturn("test");
+        when(OAuth2Util.getTenantId(anyString())).thenReturn(-1234);
+        when(OAuth2Util.getAppInformationByClientId(anyString())).thenReturn(appDO);
+        when(OAuth2Util.getClientIdForAccessToken(anyString())).thenReturn(clientID);
+
+        when(tokenValidator.validateToken(anyString())).thenReturn(tokenResponse);
+        when(userInfoEndpointConfig.getUserInfoAccessTokenValidator()).thenReturn(tokenValidator);
+        when(userInfoEndpointConfig.getUserInfoRequestValidator()).thenReturn(requestValidator);
+        mockStatic(UserInfoEndpointConfig.class);
+        when(UserInfoEndpointConfig.getInstance()).thenReturn(userInfoEndpointConfig);
+
+        Response response = openIDConnectUserEndpoint.getUserClaims(httpServletRequest);
+        assertNotNull(response.getStatus());
+        assertEquals(response.getStatus(), HttpServletResponse.SC_OK);
+
+        MultivaluedMap<String, Object> metadata = response.getMetadata();
+        String metadataValue1 = metadata.get(OAuthConstants.HTTP_RESP_HEADER_CACHE_CONTROL).toString();
+        String metadataValue2 = metadata.get(OAuthConstants.HTTP_RESP_HEADER_PRAGMA).toString();
+        assertEquals(metadataValue1, "[no-store]", "Values are not equal");
+        assertEquals(metadataValue2, "[no-cache]", "Values are not equal");
+        assertNotNull(response);
+        if (authResponse != null) {
```

Comment:
do we need this if condition?

Reply:
Yes, there have some test cases when oauth respose is null then it will provide error responses.